tests/RobotFramework/tests/cumulocity/registration/device_registration.robot

```diff
@@ -193,13 +198,55 @@ Entities persisted and restored
         Should Have MQTT Messages    c8y/s/us/plc2    message_contains=102    date_from=${timestamp}    minimum=0    maximum=0
     END
 
+Entities send to cloud on restart
+    Execute Command    tedge mqtt pub --retain 'te/factory/shop/plc1/' '{"@type":"child-device","@id":"plc1"}'
+    Execute Command    tedge mqtt pub --retain 'te/factory/shop/plc2/' '{"@type":"child-device","@id":"plc2"}'
+    Execute Command    tedge mqtt pub --retain 'te/factory/shop/plc1/sensor1' '{"@type":"child-device","@id":"plc1-sensor1","@parent":"factory/shop/plc1/"}'
+    Execute Command    tedge mqtt pub --retain 'te/factory/shop/plc1/sensor2' '{"@type":"child-device","@id":"plc1-sensor2","@parent":"factory/shop/plc1/"}'
+    Execute Command    tedge mqtt pub --retain 'te/factory/shop/plc2/sensor1' '{"@type":"child-device","@id":"plc2-sensor1","@parent":"factory/shop/plc2/"}'
+    Execute Command    tedge mqtt pub --retain 'te/factory/shop/plc1/metrics' '{"@type":"service","@id":"plc1-metrics","@parent":"factory/shop/plc1/"}'
+    Execute Command    tedge mqtt pub --retain 'te/factory/shop/plc2/metrics' '{"@type":"service","@id":"plc2-metrics","@parent":"factory/shop/plc2/"}'
+
+    External Identity Should Exist    plc1
+    External Identity Should Exist    plc2
+    External Identity Should Exist    plc1-sensor1
+    External Identity Should Exist    plc1-sensor2
+    External Identity Should Exist    plc2-sensor1
+    External Identity Should Exist    plc1-metrics
+    External Identity Should Exist    plc2-metrics
+
+    Execute Command    cat /etc/tedge/.tedge-mapper-c8y/entity_store.jsonl
+    ${original_last_modified_time}=    Execute Command    date -r /etc/tedge/.tedge-mapper-c8y/entity_store.jsonl
+
+    FOR    ${counter}    IN RANGE    0    5
+        ${timestamp}=    Get Unix Timestamp
+        Restart Service    tedge-mapper-c8y
+        Service Health Status Should Be Up    tedge-mapper-c8y
+
+        # Assert that the file contents changed on restart
+        ${last_modified_time}=    Execute Command    date -r /etc/tedge/.tedge-mapper-c8y/entity_store.jsonl
+        Should Not Be Equal    ${last_modified_time}    ${original_last_modified_time}
+        
+        # Assert that entities are sent to cloud again
+        Should Have MQTT Messages    c8y/s/us    message_contains=101,plc    date_from=${timestamp}    minimum=2    maximum=2
+        Should Have MQTT Messages    c8y/s/us/plc1    message_contains=101    date_from=${timestamp}    minimum=2    maximum=2
+        Should Have MQTT Messages    c8y/s/us/plc2    message_contains=101    date_from=${timestamp}    minimum=1    maximum=1
+        Should Have MQTT Messages    c8y/s/us/plc1    message_contains=102    date_from=${timestamp}    minimum=1    maximum=1
+        Should Have MQTT Messages    c8y/s/us/plc2    message_contains=102    date_from=${timestamp}    minimum=1    maximum=1
+
+        Sleep    2s    reason=Make sure that previous iteration was sent to cloud
```

Comment:
Why was this sleep required? That too at the end of this test. The reason isn't clear from the comment.

Reply:
I noticed that sometimes tests just fail because of that test case:
`Should Have MQTT Messages    c8y/s/us    message_contains=101,${prefix}plc    date_from=${timestamp}    minimum=2    maximum=2`. It returns more messages than expected (It can be noticed for example in [this run](https://github.com/thin-edge/thin-edge.io/actions/runs/7734517184/job/21098432434#step:8:282)). The only solution I found to this was adding this sleep, but maybe someone else has better solution to that.